provider/rfc2136/rfc2136.go

```diff
@@ -174,11 +170,6 @@ func NewRfc2136Provider(hosts []string, port int, zoneNames []string, insecure b
 
 // KeyData will return TKEY name and TSIG handle to use for followon actions with a secure connection
 func (r *rfc2136Provider) KeyData(nameserver string) (keyName string, handle *gss.Client, err error) {
-	// Check if we already have credentials for this nameserver
```

Comment:
Removed the `credentials` map to avoid issues with stale or invalid state when `ctx` is lost between runs. This simplifies the code, ensures fresh credentials are generated each time, and improves reliability by avoiding potential concurrency issues.